controllers.js

```diff
@@ -1,11 +1,31 @@
-const { selectTopics } = require("./models");
+const models = require("./models");
 
 exports.getTopics = (req, res, next) => {
-  selectTopics(req.query)
+  models
+    .selectTopics(req.query)
     .then((topics) => {
       res.status(200).send({ topics });
     })
     .catch((err) => {
       next(err);
     });
 };
+
+exports.getArticlesById = (req, res, next) => {
+  console.log(req.params.article_id);
+  models
+    .selectArticlesById(req.params.article_id)
+    .then((article) => {
+      // to catch the error here
+      if (article) {
+        res.status(200).send({ article });
+      } else {
+        res.status(404).send({
+          msg: `No article found under article_id ${req.params.article_id}`,
```

Comment:
lets use the model to check for this error and handle it using with next middleware.